src/utils/chatCommand.ts

```diff
@@ -35,41 +55,65 @@ export default function useChatCommand() {
   const [ambassadorNames] = useState(AmbassadorData.map((ambassador) => ambassador.name.split(' ')[0].toLowerCase()))
   const [diacriticsMap] = useState<Map<string, string>>(getMapOfAmbassadorWithDiacritics())
 
-  const [client] = useState(new tmi.Client({
-    connection: {
-      secure: true,
-      reconnect: true
-    },
-    channels: [
-      // 'AbdullahMorrison', //! For testing purposes
-      'Maya',
-      'AlveusSanctuary'
-    ]
-  }))
-
   const messageHandler = useCallback((channel: string, tags: ChatUserstate, msg: string, self: boolean) => {
-    //ignore if user is not a moderator or broadcaster or if the user is not AbdullahMorrison
-    if (!tags.mod && !tags.badges?.broadcaster && tags.username !== 'abdullahmorrison') return
+    //ignore if user is not a moderator or broadcaster or test user
+    if (!tags.mod && !tags.badges?.broadcaster && tags.username !== process.env.REACT_APP_CHAT_COMMANDS_PRIVILEGED_USER) return
     // Ignore echoed messages (messages sent by the bot) and messages that don't start with '!'
     if (self || !msg.trim().startsWith('!')) return
 
     const commandName = msg.trim().toLowerCase()
     if (ambassadorNames.find((name) => name === commandName.slice(1))) {
       setCommand(commandName)
     } else if (diacriticsMap.get(commandName.slice(1))) { // Check if a user typed a name without diacritics (Ex: !jalapeno should be !Jalapeño)
-      setCommand("!"+diacriticsMap.get(commandName.slice(1)))
+      setCommand("!" + diacriticsMap.get(commandName.slice(1)))
     }
   }, [ambassadorNames, diacriticsMap])
 
   const connectedHandler = useCallback(() => {
     console.log('*Twitch extension is connected to chat*')
   }, [])
 
+  const [channelInfo, setChannelInfo] = useState<ChannelInfo | null>(null)
+
   useEffect(() => {
+    let handleAuthorized: (auth: Twitch.ext.Authorized) => any = async (auth) => {
+      const newChannelInfo = await fetchCurrentChannelInfo(auth.channelId, auth)
+      if (newChannelInfo) {
+        // NOTE: We could use the channel info here to e.g. parse the title and
+        //      highlight the ambassadors for the current scene
+        setChannelInfo(newChannelInfo)
+      }
+    }
+
+    if ('Twitch' in window) {
+      window.Twitch.ext.onAuthorized(handleAuthorized)
+    }
+
+    return () => {
+      handleAuthorized = (_auth) => undefined
```

Comment:
🤔 What is this doing? Reassigning here won't do anything, as a reference to the original method has already been passed to `onAuthorized`

Reply:
1. good catch, somehow missed that i use the fn directly instead of `window.Twitch.ext.onAuthorized((auth) => handleAuthorized(auth))`
2. and I am not sure if this would even ever happen. It was supposed to "clean" up the async effect to prevent multiple calls from resolving out of order, but I guess this should never happen with this API. Using the effect with no deps (`[]`) also should pretty much guarantee it wont be called again. So probably fine to drop the cleanup fn.

Reply:
Yeah, the only reason really would be to clean up if this were to ever be unmounted -- I'd be inclined to say switching to `window.Twitch.ext.onAuthorized((auth) => handleAuthorized(auth))` is probably the better thing to do